source/azure_iot_hub_client.c

```diff
@@ -126,6 +126,29 @@ static void prvMQTTProcessSuback( AzureIoTHubClient_t * pxAzureIoTHubClient,
 }
 /*-----------------------------------------------------------*/
 
+/**
+ *
+ * Handle any incoming puback messages.
+ *
+ * */
+static void prvMQTTProcessPuback( AzureIoTHubClient_t * pxAzureIoTHubClient,
+                                  AzureIoTMQTTPacketInfo_t * pxIncomingPacket,
+                                  uint16_t usPacketID )
+{
+    ( void ) pxIncomingPacket;
+
+    configASSERT( pxIncomingPacket != NULL );
+    configASSERT( ( azureiotmqttGET_PACKET_TYPE( pxIncomingPacket->ucType ) ) == azureiotmqttPACKET_TYPE_PUBACK );
+
+    AZLogInfo( ( "Puback received for packet id: 0x%08x", usPacketID ) );
+
+    if( pxAzureIoTHubClient->_internal.xTelemetryCallback != NULL )
+    {
+        pxAzureIoTHubClient->_internal.xTelemetryCallback( usPacketID );
+    }
```

Comment:
should we add Debug log for tracing? Like entering callback and done with callback